src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/tasks/MigratedCaseLinkUpdatesTask.java

```diff
@@ -0,0 +1,196 @@
+    private static final String EVENT_ID = "migrateCaseLinkDetails";
+    @Value("${cron.caseTypeId}")
+    private String caseTypeIdsString;
+
+    @Value("${cron.maxCasesPerSearch}")
+    private int maxCases;
+    private static final int TWO = 2;
+
+    @Scheduled(cron = "${cron.updateTransferredCaseLinks}")
+    public void updateTransferredCaseLinks() {
+
+        if (!featureToggleService.isUpdateTransferredCaseLinksEnabled()) {
+            return;
+        }
+
+        String query = buildStartQuery();
+        String adminUserToken = adminUserService.getAdminUserToken();
+        String[] caseTypeIds = caseTypeIdsString.split(",");
+
+        Arrays.stream(caseTypeIds).forEach(caseTypeId -> {
+            try {
+                //Get transferred cases by case type
+                List<SubmitEvent> transferredCases = ccdClient.buildAndGetElasticSearchRequest(adminUserToken,
+                        caseTypeId, query);
+
+                for (SubmitEvent submitEvent : transferredCases) {
+                    //find possible duplicate cases by ethos reference
+                    //list of pairs of 'case type id' and 'list of submit events'
+                    List<Pair<String, List<SubmitEvent>>> listOfPairs = findCaseByEthosReference(
+                            adminUserToken, submitEvent.getCaseData().getEthosCaseReference());
+
+                    for (Pair<String, List<SubmitEvent>> pair : listOfPairs) {
+                        List<SubmitEvent> duplicates = pair.getRight();
+                        if (duplicates != null && haveSameCheckedFieldValues(duplicates)) {
+                            //update valid matching duplicates by triggering event for case
+                            String sourceCaseTypeId = pair.getLeft();
+                            triggerEventForCase(adminUserToken, submitEvent, duplicates, caseTypeId, sourceCaseTypeId);
+                        }
+                    }
+                }
+            } catch (Exception e) {
+                log.error(e.getMessage());
+            }
+        });
+    }
+
+    // Checked field values : ethos ref, respondent, claimant, submission ref(i.e. FeeGroupReference),
+    // and date of receipt
+    private boolean haveSameCheckedFieldValues(List<SubmitEvent> duplicates) {
+        SubmitEvent sourceCaseData = duplicates.get(0);
+        SubmitEvent targetCaseData = duplicates.get(1);
+        if (sourceCaseData.getCaseData() == null || targetCaseData.getCaseData() == null) {
+            return false;
+        }
+
+        boolean checkResult = sourceCaseData.getCaseData().getEthosCaseReference().equals(
+                targetCaseData.getCaseData().getEthosCaseReference())
+                && sourceCaseData.getCaseData().getClaimant().equals(
+                        targetCaseData.getCaseData().getClaimant())
+                && sourceCaseData.getCaseData().getRespondent().equals(
+                        targetCaseData.getCaseData().getRespondent())
```

Comment:
It might be worth leaving this check out because the 'lead' respondent on a case can change if the respondent is no longer continuing or has been struck out

Reply:
Makes sense.  I have adjusted it.